### PR TITLE
add schedule view for student

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/75012

## WHAT (optional)
- I add schedule viewing feature for student

## HOW
- I add interface, controller for this feature, also fix some bugs for authenticating roles.

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/student/schedule.controller.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        semesterName = SemesterNames.SECOND;
+      else semesterName = SemesterNames.FIRST;
+    }
+    if (classes.length > 0) {
+      const classId = sclass ? parseInt(sclass.toString()) : classes[0].id;
+      const isValidClassId = classes.some((_class) => _class.id === classId);
+      if (!isValidClassId) {
+        const errors = [{ msg: req.t("schedule.unauthorized") }];
+        return res.render("schedule/student-schedule", {
+          user: (req.session as CustomSessionData).user,
```
Comment (huongnt-2545) [inline src/controllers/student/schedule.controller.ts]:
```suggestion
          user,
```

File: src/controllers/student/schedule.controller.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        });
+      } else {
+        const scheduleResult = await scheduleService.getSchedules(
+          classId,
+          semesterName
+        );
+        if (typeof scheduleResult === "string") {
+          const errors = [{ msg: req.t(scheduleResult) }];
+          return res.render("schedule/student-schedule", {
+            user: (req.session as CustomSessionData).user,
```
Comment (huongnt-2545) [inline src/controllers/student/schedule.controller.ts]:
```suggestion
            user,
```

File: src/controllers/student/schedule.controller.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            classes,
+            schedule: scheduleResult,
+            errors,
+          });
+        }
+        return res.render("schedule/student-schedule", {
+          user: (req.session as CustomSessionData).user,
```
Comment (huongnt-2545) [inline src/controllers/student/schedule.controller.ts]:
```suggestion
          user,
```

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**